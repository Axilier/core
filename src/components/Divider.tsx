import React from 'react';
import { DividerProps } from '../Types';
import Layout from './Layout';

const Divider = ({ children, style }: DividerProps): JSX.Element => (
    <Layout
        orientation={'row'}
        style={{
            ...style,
            alignItems: 'center',
            width: '100%',
            fontFamily: "'IBM Plex Sans', sans-serif",
        }}
    >
        <div
            style={{
                backgroundColor: '#BEBEBE',
                width: '100%',
                height: '2px',
            }}
        />
        <div
            style={{
                padding: '0 8px',
                color: '#BEBEBE',
                verticalAlign: 'center',
            }}
        >
            {children}
        </div>
        <div
            style={{
                backgroundColor: '#BEBEBE',
                width: '100%',
                height: '2px',
            }}
        />
    </Layout>
);

export default Divider;
