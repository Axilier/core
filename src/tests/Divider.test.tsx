import React from 'react';
import { mount } from 'enzyme';
import Divider  from '../components/Divider';

const divider = mount(<Divider>test</Divider>);

describe('when the divider is used in an application', () => {
    it('should render the contained divider correctly', () => {
        expect(divider).toMatchSnapshot();
    });
});
