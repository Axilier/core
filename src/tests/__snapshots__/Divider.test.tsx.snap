// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the divider is used in an application should render the contained divider correctly 1`] = `
<Divider>
  <Layout
    orientation="row"
    style={
      Object {
        "alignItems": "center",
        "fontFamily": "'IBM Plex Sans', sans-serif",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "fontFamily": "'IBM Plex Sans', sans-serif",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#BEBEBE",
            "height": "2px",
            "width": "100%",
          }
        }
      />
      <div
        style={
          Object {
            "color": "#BEBEBE",
            "padding": "0 8px",
            "verticalAlign": "center",
          }
        }
      >
        test
      </div>
      <div
        style={
          Object {
            "backgroundColor": "#BEBEBE",
            "height": "2px",
            "width": "100%",
          }
        }
      />
    </div>
  </Layout>
</Divider>
`;
